mgowrapper.go

```diff
@@ -0,0 +1,208 @@
+package mgohttp
+
+import (
+	"context"
+	"fmt"
+
+	opentracing "github.com/opentracing/opentracing-go"
+	opentracinglog "github.com/opentracing/opentracing-go/log"
+	mgo "gopkg.in/mgo.v2"
+	bson "gopkg.in/mgo.v2/bson"
+)
+
+type tracedMgoSession struct {
+	sess *mgo.Session
+	ctx  context.Context
+}
+
+func (ts tracedMgoSession) DB(name string) MongoDatabase {
+	sp := opentracing.SpanFromContext(ts.ctx)
+	sp.LogFields(opentracinglog.String("mgo-db-name", name))
+	return tracedMgoDatabase{
+		db:  ts.sess.DB(name),
+		ctx: ts.ctx,
+	}
+}
+
+func (ts tracedMgoSession) Ping() error {
+	sp, _ := opentracing.StartSpanFromContext(ts.ctx, "ping")
+	defer sp.Finish()
+
+	err := ts.sess.Ping()
+	sp.LogFields(opentracinglog.Error(err))
```

Comment:
Looks like we can use `logAndReturnErr` here as well

Reply:
oops, can you tell which one I wrote first? 😜